connectors/migrations/20250125_microsoft_folder_parents_fix.ts

```diff
@@ -0,0 +1,99 @@
+import { concurrentExecutor, MIME_TYPES } from "@dust-tt/types";
+import _ from "lodash";
+import type { LoggerOptions } from "pino";
+import type pino from "pino";
+import { makeScript } from "scripts/helpers";
+
+import { getParents } from "@connectors/connectors/microsoft/temporal/file";
+import { dataSourceConfigFromConnector } from "@connectors/lib/api/data_source_config";
+import { upsertDataSourceFolder } from "@connectors/lib/data_sources";
+import { MicrosoftNodeModel } from "@connectors/lib/models/microsoft";
+import { ConnectorResource } from "@connectors/resources/connector_resource";
+import { MicrosoftNodeResource } from "@connectors/resources/microsoft_resource";
+
+async function backfillConnector(
+  connector: ConnectorResource,
+  execute: boolean,
+  parentLogger: pino.Logger<LoggerOptions & pino.ChildLoggerOptions>
+) {
+  const startSyncTs = Date.now();
+  const logger = parentLogger.child({
+    connectorId: connector.id,
+    execute,
+  });
+  logger.info("Starting backfill");
+
+  const dataSourceConfig = dataSourceConfigFromConnector(connector);
+  const folders = await MicrosoftNodeModel.findAll({
+    where: {
+      connectorId: connector.id,
+      nodeType: "folder",
+    },
+    attributes: ["internalId"],
+  });
+
+  const result = await concurrentExecutor(
+    folders,
+    async (folder): Promise<number> => {
+      const folderResource = await MicrosoftNodeResource.fetchByInternalId(
+        connector.id,
+        folder.internalId
+      );
+      if (!folderResource) {
+        throw new Error(
+          "Unexpected error fetching folder resource from folder model"
+        );
+      }
+      const parents = await getParents({
+        connectorId: connector.id,
+        internalId: folderResource.internalId,
+        startSyncTs,
+      });
+      if (execute) {
+        await upsertDataSourceFolder({
+          dataSourceConfig,
+          folderId: folderResource.internalId,
+          title: folderResource.name ?? "Untitled folder",
```

Comment:
can we make it consistent with activities ? "Untitled folder" id probably nicer than ""